sqlmesh/core/engine_adapter/shared.py

```diff
@@ -47,3 +55,74 @@ class DataObject(PydanticModel):
     schema_name: str = Field(alias="schema")
     name: str
     type: DataObjectType
+
+
+def set_catalog(
+    *,
+    override: t.Optional[CatalogSupport] = None,
+) -> t.Callable:
+    def decorator(func: t.Callable) -> t.Callable:
+        @functools.wraps(func)
+        def wrapper(*args: t.Any, **kwargs: t.Any) -> t.Any:
+            def _get_args_pos_and_kwarg_name(
+                func: t.Callable,
+            ) -> t.Optional[t.Tuple[str, int, str]]:
+                spec = inspect.getfullargspec(func)
+                for i, name in enumerate(spec.args):
+                    obj_type = spec.annotations.get(name)
+                    if obj_type == "SchemaName":
```

Comment:
SchemaName.__name__ ?

Reply:
`SchemaName` is part of typing and therefore generally expencted to not be imported and used during runtime. Therefore I would rather keep that than avoid using this string.